README.md

```diff
@@ -257,6 +257,49 @@ codacy-analysis-cli analyse \
 
 > In alternative to setting `--api-token` you can define CODACY_API_TOKEN in the environment.
 
+## GitHub Action
+
+The analysis CLI is available to be executed as a GitHub Action. The following is an example of a workflow
+using the CLI as an action, to analyse each commit and pull request, with your Codacy configuration.
+
+```yaml
+name: codacy-analysis-cli
+
+on: ["push"]
+
+jobs:
+  codacy-analysis-cli:
+    runs-on: ubuntu-latest
+    name: codacy-analysis-cli
+    steps:
+      - uses: actions/checkout@master
+      - name: Run codacy-analysis-cli
+        uses: codacy/codacy-analysis-cli@master
+        with:
+          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+```
+
+Running the action with the default configurations will:
+
+- Analyse the current commit or pull request, using your Codacy configurations, running all supported
+    tools for the languages you are using.
+- Print analysis results into the console (you can check them in GitHub's action workflow panel).
+- Fail the workflow if at least one issue is found in your code.
+
+Check the next section to see how you can further configure this action.
+
+### Caveats
+
+This action supports all [CLI configuration options](#commands-and-configuration) with the following exceptions:
+
+- `--codacy-api-base-url` -- **Not supported**. The action will only get your configurations from Codacy cloud.
+- `--commit-uuid` -- **Not supported**. The action will only analyse the commit that triggered it.
+- `--api-token` -- **Not supported**. Use [`--project-token`](#project-token) instead.
```

Comment:
This option (and the two below) are not supported due to the lack of conditionals in the action yaml.

For example, supporting this means that `project-token` (and these three configurations) is optional. But the way we deal with that, in the action yaml, is to have a default value instead, which in this case does not make sense.

Reply:
Did you check this https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions  or this https://github.community/t/conditional-expression-syntax/17536 ?

Reply:
Those conditionals are for workflow itself, and not the action definition. Also, I was not completely clear. Paraphrasing one person in your second link: "What I’d really like to see is jinja2 expressions allowed in the YAML though. ".

Reply:
This was solved by not using `docker` mode in GH actions to run the CLI. By using the `composite` mode we can invoke bash and use the conditionals from there.